pkg/operator/executor.go

```diff
@@ -30,39 +30,33 @@ func (e *Executor) Run(ctx context.Context) *ExecutionReport {
 
 	err = e.phaser.commit(ctx)
 	if err != nil {
-		rollbackError := e.phaser.rollback(ctx)
-		if rollbackError != nil {
-			e.currentPhase = ROLLBACK
-			return executionReportWithError(
-				wrapRollbackError(err, rollbackError),
-				e.currentPhase,
-				e.operationID,
-			)
-
-		}
-		return executionReportWithError(err, e.currentPhase, e.operationID)
+		return e.handlePhaseError(err, ctx)
 	}
 
 	e.currentPhase = VERIFY
 	err = e.phaser.verify(ctx)
 	if err != nil {
-		rollbackError := e.phaser.rollback(ctx)
-		if rollbackError != nil {
-			e.currentPhase = ROLLBACK
-			return executionReportWithError(
-				wrapRollbackError(err, rollbackError),
-				e.currentPhase,
-				e.operationID,
-			)
-		}
-		return executionReportWithError(err, e.currentPhase, e.operationID)
+		return e.handlePhaseError(err, ctx)
 	}
 
 	diff := e.phaser.operationDiff(ctx)
 
 	return executionReportWithSuccess(diff, e.currentPhase, e.operationID)
 }
 
+func (e *Executor) handlePhaseError(err error, ctx context.Context) *ExecutionReport {
+	rollbackError := e.phaser.rollback(ctx)
+	if rollbackError != nil {
+		e.currentPhase = ROLLBACK
```

Comment:
I have a doubt here: the current phase becomes `ROLLBACK` only after a failed already attempted rollback.

Wouldn't it fit better that the `ROLLBACK` phase starts before actually rolling back?

Reply:
Ok I think I see now. Thanks for clarifying.
I will keep it as it is.